packages/nx/src/native/index.js

```diff
@@ -75,10 +75,21 @@ Module._load = function (request, parent, isMain) {
     if (existsSync(tmpFile)) {
       return originalLoad.apply(this, [tmpFile, parent, isMain]);
     }
-    if (!existsSync(nativeFileCacheLocation)) {
-      mkdirSync(nativeFileCacheLocation, { recursive: true });
+
+    mkdirSync(nativeFileCacheLocation, { recursive: true });
+
+    try {
+      copyFileSync(nativeLocation, tmpFile, constants.COPYFILE_EXCL);
```

Comment:
This can still be a race condition, because the other process may not have finished writing the file by the time node tries to read it. I could add a loop to append an incrementing number to the file name until success or max retries to mitigate this, but that's not ideal since it creates extra garbage.